sympy/solvers/solvers.py

```diff
@@ -1120,41 +1146,82 @@ def _has_piecewise(e):
     # try to get a solution
     ###########################################################################
     if bare_f:
-        solution = _solve(f[0], *symbols, **flags)
-        # solution is:
-        # dict for coeficient system with one solution
-        # list of values
-        # list of dicts
+        solution = None
+        # check for undetermined coefficients situation: `solve(a*x + 2*x + b - c, a, b)`
+        if len(symbols) != 1:
```

Comment:
Everything inside this block should be moved to a separate function.

Reply:
Thanks for the suggestions for handling the flags. That and the removal of the special handling for solving undetermined coefficients has been made.